python/lsst/dax/data_generator/columns.py

```diff
@@ -517,3 +514,150 @@ def __call__(self, box, length, seed, prereq_row=None, prereq_tables=None, uniqu
+        assert prereq_tables is not None, "SourceGenerator requires the Visit table."
+        assert box_center is not None, "Must supply box center"
+
+        np.random.seed(calcSeedFrom(unique_box_id, seed, self.column_seed))
+
+        visit_table = prereq_tables['CcdVisit']
+        object_table = prereq_tables['Object']
+
+        objects_inside_box = object_table[(object_table['psRa'] >= box.raA) &
+                                          (object_table['psRa'] < box.raB) &
+                                          (object_table['psDecl'] >= box.decA) &
+                                          (object_table['psDecl'] < box.decB)]
+        v_radius = self.visit_radius * 1.5  # Go a little bit bigger so nothing is missed.
+        min_dec = box.decA - v_radius
+        max_dec = box.decB + v_radius
+        trimmed_visit = visit_table.loc[(visit_table['decl'] >= min_dec) & (visit_table['decl'] <= max_dec)]
+        print(f"edge_only={edge_only} len trimmed={len(trimmed_visit)}  base={len(visit_table)}")
+
+        # Unlike the ForcedSource table, only some of the objects should have visits, not
+        # all of them. It would be nice to have the brightest objects get visits, but that
+        # isn't that important right now. Just trim objects from objects_inside_box until
+        # the right ratio is reached for now.
+        target_percentage = 0.3  # supplied by Colin
+        df = objects_inside_box.copy()
+        sel, = np.where(np.random.random(len(objects_inside_box)) > 1.0 - target_percentage)
+        df = objects_inside_box.iloc[sel]
+        objects_inside_box = df
+
+        visit_skycoords = SkyCoord(ra=trimmed_visit['ra'], dec=trimmed_visit['decl'], unit="deg")
+        visit_deltas = box_center.separation(visit_skycoords).degree
+        sel_matching_visits, = np.where(visit_deltas < self.visit_radius)
+        n_matching_visits = len(sel_matching_visits)
+        print(f"Found {n_matching_visits} matching visits")
+
+        out_objectIds = np.repeat(objects_inside_box['objectId'].values, n_matching_visits)
+        out_ccdVisitIds = np.tile(trimmed_visit['ccdVisitId'].iloc[sel_matching_visits].values,
+                                  len(objects_inside_box))
+        # TODO: Figure out a way to get RA and Dec column names from  RaDecGenerator
+        out_obj_ras = np.repeat(objects_inside_box['psRa'].values, n_matching_visits)
+        out_obj_decs = np.repeat(objects_inside_box['psDecl'].values, n_matching_visits)
+
+        # Use the string of Source columns to generate the needed columns.
+        # TODO: maybe allow values in parenthesis to indicate max, min, etc
+        spec_c = spec_cols.split(",")
+        col_dict = {}
+        indx = 0
+        for cn in spec_c:
+            col_info = {}
+            cname, ctype = cn.split(":")
+            col_info['cname'] = cname
+            col_info['ctype'] = ctype
+            col_info['data'] = None
+            col_dict[indx] = col_info
```

Comment:
`col_dict` is effectively just a list, so this can be
````
col_list = []
for cn in spec_c:
    cname, ctype = cn.split(":")
    col_list.append({"cname": cname, "ctype": ctype, "data": None})
````